compiler/ir/ir.objcinterop/src/org/jetbrains/kotlin/ir/objcinterop/ObjCInterop.kt

```diff
@@ -266,9 +266,12 @@ fun IrClass.getExternalObjCClassBinaryName(): String =
         this.getExplicitExternalObjCClassBinaryName()
                 ?: this.name.asString()
 
-fun IrClass.getExternalObjCMetaClassBinaryName(): String =
-        this.getExplicitExternalObjCClassBinaryName()
-                ?: this.name.asString().removeSuffix("Meta")
+fun IrClass.getExternalObjCMetaClassBinaryName(): String {
+    this.getExplicitExternalObjCClassBinaryName()?.let { return it }
+    // External ObjC metaclass is named as `Foo.Componion`, Foo is subclass of NSObject or itself.
+    if (this.isCompanion) return (this.parent as IrClass).getExternalObjCClassBinaryName()
```

Comment:
This addition probably is not used anymore.
For example, the function is no longer used for a `companion` when generating a type check. Also, if it is used for a `companion` somewhere, it will just crash, because `getExplicitExternalObjCClassBinaryName` will throw NPE, as before.

Please revert the change.